test/integration/resourcemanager/managedresource/resource_test.go

```diff
@@ -977,6 +985,58 @@ var _ = Describe("ManagedResource controller tests", func() {
 			})
 		})
 	})
+
+	Describe("Immutable resources", func() {
+		It("should recreate the resource if the update is invalid", func() {
+			configMap.Immutable = pointer.Bool(true)
+			secretForManagedResource = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      resourceName,
+					Namespace: testNamespace.Name,
+				},
+				Data: secretDataForObject(configMap, dataKey),
+			}
+
+			Eventually(func(g Gomega) []gardencorev1beta1.Condition {
+				g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+				return managedResource.Status.Conditions
+			}).Should(
+				ContainCondition(OfType(resourcesv1alpha1.ResourcesApplied), WithStatus(gardencorev1beta1.ConditionTrue), WithReason(resourcesv1alpha1.ConditionApplySucceeded)),
+			)
+
+			Expect(testClient.Delete(ctx, configMap)).To(Succeed())
+
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap)
+			}).Should(BeNotFoundError())
+
+			newData := map[string]string{
+				"foo": "bar",
+			}
+			oldData := configMap.Data
+			configMap.Data = newData
+			configMap.Immutable = pointer.Bool(true)
```

Comment:
Now that the immutable field is set at the start. I just wanted to let you know that this line is not needed.
```suggestion
```